oslabs-beta/Kubiquity
# Set up structure for Kubiquity

WHAT DOES THIS PULL REQUEST DO?
- Creates basic structure for Kubiquity app based (composed with Electron and React). 

HOW TO TEST THIS PR:
IF YOU HAVE NOT ALREADY FORKED THE REPO:
- Fork: https://github.com/oslabs-beta/Kubiquity
- Once forked, in your terminal, run `git clone https://github.com/[yr Github username]/Kubiquity`. 
- Run `git remote add upstream https://github.com/oslabs-beta/Kubiquity`. 

ONCE THE REPO IS FORKED:
- Open this PR:
![Screen Shot 2021-07-01 at 1 07 22 PM](https://user-images.githubusercontent.com/44009893/124163212-5a86b480-da6d-11eb-85a7-4f4731acdd49.png) 
- In your terminal, run `npm install && npm run watch`. The second command will run a webpack build. Leave Webpack running so that we have access to React. 
- In a second window (concurrently), run `npm start`. This comment will run Electron. 
- An Electron app should pop up; verify that it says our app's name and some placeholder text. 
- Make a change to the React component; verify that the Electron app hot-reloads without any refresh. 

ONCE YOU'VE VERIFIED THAT THE APP WORKS FOR YOU:
- Look over the code via Github, and leave any comments for questions / concerns / changes. 

WHAT IS NOT ADDRESSED HERE:
- We likely will do some additional work to configure our app to both work in production and to have a backend. There likely will have to be some additional Electron-needed changes to our structure and packages; however, we can cross those bridges when they arise. 
- We should add a dev script that runs our electron and webpack scripts at the same time rather than using two terminals.
